File: web-client/src/types/models/WinningTile.ts
import RectangularTile from "@/types/models/RectangularTile.ts";
import {TileVisitor} from "@/types/visitors/TileVisitor.ts";

/**
 * Rappresenta una tessera vincente, che estende la classe RectangularTile.
 */
export default class WinningTile extends RectangularTile {

    /**
     * Accetta un visitor per la tessera vincente.
     * @param {TileVisitor} visitor - Il visitor da accettare.
     */
    accept(visitor: TileVisitor) {
        visitor.visitWinningTile(this)
    }
}

File: solutions-configurer/src/it/klotski/solutions/configurer/WinningTile.java
package it.klotski.solutions.configurer;

/**
 * Questa classe rappresenta una tessera vincente nel gioco.
 * Estende la classe RectangularTile e implementa l'interfaccia IWinningTile.
 */
public class WinningTile extends RectangularTile {

    /**
     * Costruttore per la classe WinningTile.
     *
     * @param id     l'id della tessera
     * @param x      la posizione x della tessera
     * @param y      la posizione y della tessera
     * @param width  la larghezza della tessera
     * @param height l'altezza della tessera
     */
    public WinningTile(int id, int x, int y, int width, int height) {
        super(id, x, y, width, height);
    }

}